manager/list_view.py

```diff
@@ -36,34 +47,30 @@ def _apply_args(self, args):  # pylint: disable=R0912, R0915
             self.query = self.query.where(expressions)
         self.active_filter = args["filter"]
 
-        self.active_sort = None
-        sort_column_names = []
+        input_sort_columns = []
         if args["sort"]:
-            if not self.sortable_columns:
-                raise InvalidArgumentException("No available columns for sorting")
-            input_sort_columns = args["sort"].split(",")
-            sort_columns = []
-            for column_name in input_sort_columns:
-                desc = False
-                if column_name.startswith("-"):
-                    desc = True
-                    column_name = column_name[1:]
-                elif column_name.startswith("+"):
-                    column_name = column_name[1:]
-                column = self.sortable_columns.get(column_name, None)
-                if column:
-                    if desc:
-                        sort_column_names.append("-%s" % column_name)
-                        sort_columns.append(column.desc(nulls='LAST'))
-                    else:
-                        sort_column_names.append("+%s" % column_name)
-                        sort_columns.append(column.asc())
+            input_sort_columns.extend(args["sort"].split(","))
+        if self.default_sort_columns:
+            input_sort_columns.extend(self.default_sort_columns)
+
+        if input_sort_columns and not self.sortable_columns:
+            raise InvalidArgumentException("No available columns for sorting")
+
+        sort_columns = []
+        for column_name in input_sort_columns:
+            column_name, desc = self._parse_column_name(column_name)
+            column = self.sortable_columns.get(column_name, None)
+            if column:
+                if desc:
+                    sort_columns.append(column.desc(nulls='LAST'))
                 else:
-                    raise InvalidArgumentException("Invalid sort column: %s" % column_name)
-            if sort_columns:
-                self.query = self.query.order_by(*sort_columns)
-            self.active_sort = ",".join(sort_column_names)
+                    sort_columns.append(column.asc())
+            else:
+                raise InvalidArgumentException("Invalid sort column: %s" % column_name)
+        if sort_columns:
+            self.query = self.query.order_by(*sort_columns)
 
+        self.active_sort = args["sort"] if args["sort"] else None
```

Comment:
This method is quite confusing to me. Maybe it would be better to create separate methods to apply particular args. I would also validate arguments at the beginning of method if possible. Is it really necessary to disable pylint rules R0912, R0915?

Reply:
I did a re-factoring fixing all pylint disables in ListView class.

Reply:
Methods are splitted to smaller ones. Validation of arguments was already done.